src/main/java/org/ow2/proactive/catalogclient/service/SpringConfig.java

```diff
@@ -23,19 +23,23 @@
  * If needed, contact us to obtain a release under GPL Version 2 or 3
  * or a different license than the AGPL.
  */
-package org.ow2.proactive.exception;
+package org.ow2.proactive.catalogclient.service;
 
-public class EntityTooLargeException extends RuntimeException {
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.annotation.PropertySources;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
 
-    public EntityTooLargeException() {
-        super();
-    }
 
-    public EntityTooLargeException(String message) {
-        super(message);
-    }
+@Configuration
+@ComponentScan(basePackages = { "org.ow2.proactive.catalogclient.*" })
+@PropertySources({ @PropertySource("classpath:catalogclient/application.properties") })
+public class SpringConfig {
```

Comment:
After looking at the issue it seems that sonar misunderstood SpringConfig class role. Sonar consider SpringConfig class as a service class whereas it is a component class.